neil

```diff
@@ -225,38 +239,41 @@
    [borkdude.rewrite-edn :as r]
    [clojure.edn :as edn]))
 
-(defn resolve-deps-file [dir deps-file]
-  (if dir
-    (fs/file dir deps-file)
-    deps-file))
+(defn resolve-deps-file 
+  "Returns a file object for `deps-file` in the `dir` directory if that file exists, else nil."
+  [dir deps-file]
+  (let [resolved (if dir
+                   (fs/file dir deps-file)
+                   (fs/file deps-file))]
+    (when (fs/exists? resolved) resolved)))
```

Comment:
I'm not sure what the consequences of this change is. One feature of neil is that when you run `neil dep add whatever` and you don't have an `deps.edn` file, that it creates one for you. At least, I've been relying on this feature myself and I'd like to keep that

Reply:
`neil dep add` calls `ensure-deps-file`, which spits to a deps file if it doesn't exist, and from what I can tell, this function doesn't come up in that particular flow.

I thought I'd found the usage and covered then with the `when-let`, but I (well, really Cursive) found a call from the `version` namespace in a new function of the same name. The usages of the `version` function mostly seem to be slurping the file, so they would also throw when used with a missing deps file.

before change:
```
$ neil version major 3
----- Error --------------------------------------------------------------------
Type:     java.io.FileNotFoundException
Message:  /home/bob/Documents/code/clojure/bbt2/deps.edn (No such file or directory)
Location: /home/linuxbrew/.linuxbrew/bin/neil:854:31
```

after change:
```
$ ../neil/neil version major 3
----- Error --------------------------------------------------------------------
Type:     java.lang.NullPointerException
Location: /home/bob/Documents/code/clojure/bbt2/../neil/neil:695:3
```

I'll circle back on this and make the exists checks more local to avoid the NPE.

Reply:
It gets even more fun... `version` also calls the calling functions, so there's a wider-spread assumption that a deps file will be present, and there are cases where we want to not fail (using a `new` template that doesn't have a deps.edn file), and cases where we probably should fail (trying to update the version in a missing deps file).

I'm gonna play around with some possibilities here... maybe an arg to these functions that says whether to fail on a missing file or not, maybe some modification of the function. I'll probably close this PR for a little while if I don't find an outcome I like today (I'd rather do nothing than do a bad thing).

Reply:
Thanks for getting into the weeds on this!

Reply:
Since the missing deps file is not a problem only on the `new` code path, it looks like the most straightforward change is to just check for the deps file existence on that code path.

Reply:
Makes sense

Reply:
I added a test that covers the deps file existing and not existing. A minimal deps-new template came along with it, but I think that could potentially be useful in future tests of `new`.